src/replay/consensus.zig

```diff
@@ -523,10 +526,43 @@ fn computeBankStats(
             fork_stats.computed = true;
             try new_stats.append(allocator, slot);
         }
+        try cacheTowerStats(
+            allocator,
+            progress,
+            replay_tower,
+            slot,
+            ancestors,
+        );
     }
     return try new_stats.toOwnedSlice(allocator);
 }
 
+fn cacheTowerStats(
+    allocator: std.mem.Allocator,
+    progress: *ProgressMap,
+    replay_tower: *const ReplayTower,
+    slot: Slot,
+    ancestors: *const std.AutoArrayHashMapUnmanaged(Slot, SortedSet(Slot)),
+) !void {
+    var stats = progress.getForkStats(slot) orelse return error.MissingSlot;
+
+    const slice = try replay_tower.checkVoteStakeThresholds(
+        allocator,
+        slot,
+        &stats.voted_stakes,
+        stats.total_stake,
+    );
+    defer allocator.free(slice);
+    stats.vote_threshold = try VoteThreshold.initCapacity(allocator, slice.len);
+    try stats.vote_threshold.appendSlice(allocator, slice);
```

Comment:
unnecessary allocation

```suggestion
    stats.vote_threshold = VoteThreshold.fromOwnedSlice(slice);
```

you'll also need to change the return type of checkVoteStakeThresholds to a mut slice

Reply:
[e4e6510](https://github.com/Syndica/sig/pull/851/commits/e4e651048f4b970664260248c2a98a78eca74ad2)